apm-agent-core/src/main/java/co/elastic/apm/agent/util/ClassLoaderUtils.java

```diff
@@ -52,4 +34,14 @@ public static boolean isAgentClassLoader(@Nullable ClassLoader classLoader) {
     public static boolean isBootstrapClassLoader(@Nullable ClassLoader classLoader) {
         return classLoader == null;
     }
+
+    public static boolean isPluginClassLoader(@Nullable ClassLoader classLoader) {
+        if(classLoader == null){
+            return false;
+        }
+        String clName = classLoader.getClass().getName();
+        return (IndyPluginClassLoader.class.getName().equals(clName)
+            || ExternalPluginClassLoader.class.getName().equals(clName));
```

Comment:
I think we still want to instrument classes within an external plugin. Otherwise, the public API wouldn't work in the context of an external plugin.

Reply:
Yes, we do, so also this method should be called `isInternalPluginClassLoader`

Reply:
https://github.com/elastic/apm-agent-java/pull/2376#issue-1090341350 is what I vaguely remembered - so it seems there was indeed an issue with instrumenting classes that were loaded by the IndyPlugingCL.
The part that makes sure we load he API classes through the External plugin CL, so that we can instrument them is: https://github.com/elastic/apm-agent-java/blob/c8e7936b1a6a897aef609b820d20e254cd1440cc/apm-agent-core/src/main/java/co/elastic/apm/agent/bci/IndyBootstrap.java#L383-L388